manager/integration/tests/test_node.py

```diff
@@ -368,7 +382,56 @@ def test_disable_scheduling_on_cordoned_node(client):  # NOQA
     7. Automatically the scheduler should creates three replicas
     from step 5 failure.
     """
-    pass
+    # Set `Disable Scheduling On Cordoned Node` to true
+    disable_scheduling_on_cordoned_node_setting = \
+        client.by_id_setting(SETTING_DISABLE_SCHEDULING_ON_CORDONED_NODE)
+    client.update(disable_scheduling_on_cordoned_node_setting, value="true")
+
+    # Set `Replica Node Level Soft Anti-Affinity` to false
+    node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    client.update(node_soft_anti_affinity_setting, value="false")
+
+    # Get one node
+    nodes = client.list_node()
+    node = nodes[0]
+
+    # Set cordon on node
+    set_node_cordon(core_api, node.name, True)
+
+    node = wait_for_node_schedulable_condition(
+        get_longhorn_api_client(), node.name)
+
+    # Node schedulable condition should be faulse
+    assert node.conditions[NODE_CONDITION_SCHEDULABLE]["status"] ==  \
+        "False"
+    assert node.conditions[NODE_CONDITION_SCHEDULABLE]["reason"] == \
+        "KubernetesNodeCordoned"
+
+    # Create a volume and check its schedulable condition should be false
+    vol_name = common.generate_volume_name()
+    client.create_volume(name=vol_name, size=str(SMALL_DISK_SIZE),
+                         numberOfReplicas=len(nodes))
+    common.wait_for_volume_detached(client, vol_name)
+    common.wait_for_volume_condition_scheduled(client, vol_name,
+                                               "status",
+                                               CONDITION_STATUS_FALSE)
+
+    # Set uncordon on node
+    set_node_cordon(core_api, node.name, False)
+
+    # Node schedulable condition should be true
+    node = wait_for_node_schedulable_condition(
+        get_longhorn_api_client(), node.name)
+    assert node.conditions[NODE_CONDITION_SCHEDULABLE]["status"] == "True"
+
+    # Created volume schedulable condition change to true
+    common.wait_for_volume_condition_scheduled(client, vol_name,
+                                               "status",
+                                               CONDITION_STATUS_TRUE)
+
+    # Disk storage available test on this node
+    node_controller_sync_storage_available_test(client, get_self_host_id())
```

Comment:
You can test the volume instead of the disk, e.g. use `create_and_check_volume`. The disk should be fine.

Reply:
@yasker , I realize I have create a volume, is it ok to use that volume to attach it to a node? Also do we need to use `write_volume_random_data` to write data on this volume as well?

Thanks

Reply:
Yes, we need to attach the volume to the current node before testing it.